backend/src/utils/userUtils.ts

```diff
@@ -90,7 +94,12 @@ export const getUserName = async (
     if (DEV_MODE) {
       return (currentUser.username || currentUser.name)?.split('/')[0];
     }
-    fastify.log.error(`Error retrieving username: ${e.response?.data?.message || e.message}`);
-    throw new Error(`Failed to retrieve username: ${e.response?.data?.message || e.message}`);
+    fastify.log.error(`Failed to retrieve username: ${e.response?.data?.message || e.message}`);
+    const error = createCustomError(
+      'Unauthorized',
+      `Failed to retrieve username: ${e.response?.data?.message || e.message}`,
+      e.statusCode || 500,
```

Comment:
I do not believe `statusCode` is available, is it? I thought Fastify added that on the way out 🤔

Reply:
mmmm i think the kube call adds a statusCode, at least i was intercepting statusCode when the call failed.